File: dev-repo/opportunityplus-260506_dev-deploy/UNOPS.PAO.Models/Contacts/ContactModel.cs
using System.Text.Json.Serialization;
using UNOPS.PAO.Models.Documents;
using UNOPS.PAO.Models.Interactions;
using UNOPS.PAO.Models.OrganizationUnits;
using UNOPS.PAO.Models.Shared;

namespace UNOPS.PAO.Models.Contacts;

public class ContactModel
{
    public int Id { get; set; }
    public string? Salutation { get; set; }
    public string? FirstName { get; set; }
    public string? MiddleName { get; set; }
    public string LastName { get; set; } = null!;
    public string? Suffix { get; set; }
    public string Title { get; set; } = null!;
    public string? Department { get; set; }
    public string? Description { get; set; }
    public string Email { get; set; } = null!;
    public string? Phone { get; set; }
    public string? Mobile { get; set; }
    public string? Assistant { get; set; }
    public string? AssistantPhone { get; set; }
    public string? AssistantEmail { get; set; }
    public string? MailingStreet { get; set; }
    public string? MailingStreet2 { get; set; }
    public string? MailingCity { get; set; }
    public string? MailingStateProvince { get; set; }
    public string? MailingPostalCode { get; set; }
    public string? MailingCountry { get; set; }
    public string? ProfilePictureUrl { get; set; }
    public string? Status { get; set; }
    
    /// <summary>
    /// Full name constructed from FirstName, MiddleName, and LastName
    /// </summary>
    public string FullName 
    { 
        get 
        {
            var parts = new[] { FirstName, MiddleName, LastName }
                .Where(part => !string.IsNullOrWhiteSpace(part))
                .ToArray();
            return parts.Length > 0 ? string.Join(" ", parts) : "";
        }
    }
    
    public PartnerSummaryModel? Partner { get; set; }
    public List<DocumentModel>? Documents { get; set; }

    public string? CreatedByName { get; set; }
    public string? CreatedByOfficeName { get; set; }
    
    /// <summary>
    /// Interactions associated with this contact
    /// </summary>
    public List<InteractionModel>? Interactions { get; set; }
    
    /// <summary>
    /// Permissions for this specific contact
    /// </summary>
    public EntityPermissionsModel? Permissions { get; set; }
    
    // Audit fields from ModifiableDeletableEntity (read-only from frontend perspective)
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
    public DateTime? CreatedDate { get; set; }
    
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
    public DateTime? LastModifiedDate { get; set; }
    
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public int? CreatedBy { get; set; }
    
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public int? LastModifiedBy { get; set; }

    /// <summary>Contact office scope (from <c>OfficeRelationship</c>), serialized as <c>officeRelationships</c>.</summary>
    [JsonPropertyName("officeRelationships")]
    public List<OrganizationUnitRelationshipModel>? OfficeRelationships { get; set; }
}

/// <summary>
/// Simplified partner model for contact references
/// </summary>
public class PartnerSummaryModel
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
}

File: dev-repo/opportunityplus-260506_dev-deploy/UNOPS.PAO.ClientApp/src/app/features/partnerships/contacts/models/contact.model.ts
import {Partner} from '@partnerships/partners/models/partner.model';
import { EntityPermissionSet } from '@shared/models/shared-types';
import { OrganizationUnitRelationshipModel } from '@partnerships/partners/models/organization-unit-relationship.model';
import { OrganizationHierarchyModel } from '@core/models/organization-hierarchy.model';

export interface Contact {
  id?: string | null;
  salutation?: string | null;
  firstName?: string | null;
  middleName?: string | null;
  lastName?: string | null;
  suffix?: string | null;
  title?: string | null;
  pronouns?: string | null;
  birthDate?: Date | null;
  
  email?: string | null;
  phone?: string | null;
  mobile?: string | null;
  otherPhone?: string | null;
  fax?: string | null;
  
  partner?: Partner | null;
  /** Flat partner name from list/API payloads when `partner` is not expanded */
  partnerName?: string | null;
  department?: string | null;
  description?: string | null;
  status?: string | null;
  contactNumber?: string | null;
  
  assistant?: string | null;
  assistantPhone?: string | null;
  assistantEmail?: string | null;
  
  mailingStreet?: string | null;
  mailingStreet2?: string | null;
  mailingCity?: string | null;
  mailingStateProvince?: string | null;
  mailingPostalCode?: string | null;
  mailingCountry?: string | null;
  
  profilePictureUrl?: string|null;

  discriminator?: string | null;
  createdBy?: string | null;
  createdDate?: Date | null;
  lastModifiedBy?: string | null;
  lastModifiedDate?: Date | null;
  isDeleted?: boolean | null;
  deletedBy?: string | null;
  deletedDate?: Date | null;
  
  // Import-specific properties
  isImportEdit?: boolean;
  _updated?: boolean;
  _importRowId?: string;
  
  // RBAC permissions
  permissions?: EntityPermissionSet;
  
  /** Office links (API); legacy key was organizationUnitRelationships */
  officeRelationships?: OrganizationUnitRelationshipModel[];
  organizationUnitRelationships?: OrganizationUnitRelationshipModel[];
}

/** Resolves office/org links from current or legacy API shape */
export function getContactOfficeRelationships(
  contact: Contact | null | undefined
): OrganizationUnitRelationshipModel[] | null | undefined {
  return contact?.officeRelationships ?? contact?.organizationUnitRelationships;
}

// Utility function to get the primary organization unit from a contact
export function getPrimaryOrganizationUnit(contact: Contact): OrganizationHierarchyModel | null {
  return getContactOfficeRelationships(contact)?.[0]?.organizationHierarchy || null;
}
